frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx

```diff
@@ -193,30 +195,70 @@ type LegacyDataPickerProps = {
   stageIndex: number;
   table: Lib.TableMetadata | Lib.CardMetadata | undefined;
   placeholder: string;
+  canChangeDatabase: boolean;
   hasMetrics: boolean;
   isDisabled: boolean;
   onChange: (tableId: TableId) => void;
 };
 
-function LegacyDataPicker({
+function EmbeddingDataPicker({
   query,
   stageIndex,
   table,
   placeholder,
+  canChangeDatabase,
   hasMetrics,
   isDisabled,
   onChange,
 }: LegacyDataPickerProps) {
+  const { data: dataSourceCountData, isLoading: isDataSourceCountLoading } =
+    useSearchQuery({
+      models: ["dataset", "table"],
+      limit: 0,
+    });
+
   const databaseId = Lib.databaseID(query);
   const tableInfo =
     table != null ? Lib.displayInfo(query, stageIndex, table) : undefined;
+  /**
+   * We try to blur the line between models and tables for embedding users.
+   * this property will change the way icons are displayed in the data picker trigger,
+   * so we need to remove it. Treating it as a table.
+   */
+  const tableInfoWithTableIcon = tableInfo && {
+    ...tableInfo,
+    isModel: false,
+  };
   const pickerInfo = table != null ? Lib.pickerInfo(query, table) : undefined;
   const { data: card } = useGetCardQuery(
     pickerInfo?.cardId != null ? { id: pickerInfo.cardId } : skipToken,
   );
   const context = useNotebookContext();
   const modelList = getModelFilterList(context, hasMetrics);
 
+  if (isDataSourceCountLoading) {
+    return null;
```

Comment:
I tried rendering `DatPickerTarget` directly, but it failed some tests, because some unit tests will try to get the component and click it instantly, but just `DataPickerTarget` alone isn't clickable.